test/selenium/lib/make_suite.js

```diff
@@ -3,33 +3,50 @@
 const FoxboxProcessManager = require('./foxbox_process_manager');
 const App = require('./setup_webapp');
 
-function makeSuite(description, hostUrl, subSuite) {
+function Suite(description, hostUrl) {
+  hostUrl = hostUrl || FoxboxProcessManager.HOST_URL;
+  this.app = new App(hostUrl);
 
-  if (typeof hostUrl === 'function') {
-    subSuite = hostUrl;
-    hostUrl = FoxboxProcessManager.HOST_URL;
-  }
-
-  var foxboxProcessManager = new FoxboxProcessManager();
-  var app = new App(hostUrl);
+  this.description = description;
+  this.foxboxProcessManager = new FoxboxProcessManager();
+}
 
-  describe(description, function () {
-    this.timeout(30000);
+Suite.prototype = {
+  build(subSuite) {
+    var self = this;
 
-    before(() => foxboxProcessManager.start());
+    describe(this.description, function() {
+      this.timeout(30000);
 
-    beforeEach(() => app.init());
+      before(() => {
+        return self.foxboxProcessManager.start()
+        .then(() => self.app.init());
+      });
 
-    subSuite(app);
+      subSuite(self.app);
 
-    after(() => {
-      return app.stop().then(() => {
-        foxboxProcessManager.kill();
-        return foxboxProcessManager.cleanData();
+      after(() => {
+        return self.app.stop()
+        .then(() => self.foxboxProcessManager.kill())
+        .then(() => self.foxboxProcessManager.cleanData());
       });
     });
-  });
-}
+  },
+
+  browserCleanUp() {
+    return this.app.clear()
+    .then(() => this.app.init(),
+          () => this.app.init());
+  },
```

Comment:
just curious, why the init() called twice here? i'm not so familiar with the arrow notation.